code/modules/projectiles/updated_projectiles/projectile.dm

```diff
@@ -332,7 +332,7 @@
 
 	var/distance = P.distance_travelled - 1
 
-	if(distance < 1)
+	if(P.ammo.barricade_clear_distance > distance)
```

Comment:
To follow your logic here you want a `=>` instead.

Reply:
???

Default clear distance allowance is 1.

Distance traveled is always distance traveled - 1.

Most projectiles are intended to ignore barricades 1 tile away, so this works: their barricade clear distance will be 1 and the distance after mods, will be 0. 1 is greater than 0 so the bypass will trigger.


Or did you want this to be reformatted, such that distance = P.distance traveled with >= used instead?

Reply:
Ah, I missed the -1 there. Not sure why it's there in any case, you could simplify it with `if(P.distance_travelled <= P.ammo.barricade_clear_distance)`.

Reply:
I was following existing formatting.

Reply:
All good. In this case `distance` is a variable and `P.ammo.barricade_clear_distance` a number. Please alter it to the `variable` `operator` `number` format as per the contribution guide (that would mean a `if(distance < P.ammo.barricade_clear_distance)`)